pycparser/c_parser.py

```diff
@@ -557,9 +557,9 @@ def p_pppragma_directive(self, p):
                                     | PPPRAGMA PPPRAGMASTR
         """
         if len(p) == 3:
-            p[0] = c_ast.Pragma(p[2])
+            p[0] = c_ast.Pragma(p[2], self._coord(p.lineno(2)))
         else:
-            p[0] = c_ast.Pragma("")
+            p[0] = c_ast.Pragma("", self._coord(p.lineno(2)))
```

Comment:
This looks weird - are you sure it's lineno(2) and not lineno(1) in this case? Actually, maybe both cases...

[a test for this case (both with and without string) would be good too]